src/NServiceBus.Gateway/Installer/GatewayHttpListenerInstaller.cs

```diff
@@ -1,126 +1,98 @@
-namespace NServiceBus.Installation
-{
-    using System;
-    using System.Diagnostics;
-    using System.IO;
-    using System.Runtime.InteropServices;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Gateway.Installer;
-    using Gateway.Receiving;
-    using Logging;
-    using Settings;
+namespace NServiceBus.Installation;
 
-    class GatewayHttpListenerInstaller : INeedToInstallSomething
-    {
-        IManageReceiveChannels channelManager;
-        bool enabled;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Gateway.Receiving;
+using Logging;
 
-        public GatewayHttpListenerInstaller(IReadOnlySettings settings)
+class GatewayHttpListenerInstaller(IManageReceiveChannels channelManager) : INeedToInstallSomething
```

Comment:
I think we should add a platform guard here on the class. See https://github.com/Particular/MonitoringDemo/pull/200/files#diff-0afce02e620fc4bcf734778a930b8b0992273b4e03a1efdf72fcf1398b5b04c6R138 or

Reply:
We only enable the installer if we are on Windows. Do we still need a guard here?

Reply:
makes sense